### DISCUSSION TITLE
Curl works, but HTTP doesn't

### DISCUSSION THREAD

**Discussion by Nosferican** [community/community#24884] [Apps, API and Webhooks]
<p>For example,</p>
<pre><code>computer:~ user$ curl -H "Authorization: bearer token" -X POST -d " \

&gt; { \

&gt; \"query\": \"query { viewer { login }}\" \

&gt; } \

&gt; " https://api.github.com/graphql

{"data":{"viewer":{"login":"UserName"}}}
</code></pre>
<p>However, when I attempt the same query using HTTP</p>
<pre><code>{ "data": null, "errors": [{ "message": "A query attribute must be specified and must be a string." }] }
</code></pre>

**Comment by fregante** [comment]
<p>If you’re talking about httpie, use <a href="https://github.com/jakubroztocil/httpie#redirected-input" rel="nofollow noopener">piping</a>:</p>
<pre><code>echo '{"query": "{viewer{login}}"}' | http POST https://api.github.com/graphql Authorization:"bearer xxxxxxxxxxxxxx"
</code></pre>

**Comment by Nosferican** [comment, marked as answer]
<p>It seems there were a couple issues on my end. For example, one difference was that cURL automatically adds the <code>User-Agent</code> which one has to add manually using HTTP. I couldn’t figure out the issue with R’s http, but after making sure the query were properly escaped with HTTP.jl I got it to work.</p>
